tests/test_genomes.py

```diff
@@ -158,38 +203,51 @@ def test_is_neutral(self):
 
 
 class TestGeneConversion(object):
-    species = stdpopsim.get_species("EscCol")
+    def test_mean_gene_conversion(self):
+        dro_mel = stdpopsim.get_species("DroMel")
+        contig = dro_mel.get_contig(length=1000, use_species_gene_conversion=False)
+        assert contig.gene_conversion_fraction is None
+        assert contig.gene_conversion_length is None
+        contig = dro_mel.get_contig(length=1000, use_species_gene_conversion=True)
+        mean_gc = dro_mel.genome.mean_gene_conversion_fraction
+        assert np.isclose(mean_gc, contig.gene_conversion_fraction)
+
+    def test_no_mean_gene_conversion(self):
+        # this will need to be changed if we add GC rates to AnaPla
+        ana_pla = stdpopsim.get_species("AnaPla")
+        contig = ana_pla.get_contig(length=1000, use_species_gene_conversion=True)
+        mean_gc = ana_pla.genome.mean_gene_conversion_fraction
+        assert mean_gc == 0.0
+        assert contig.gene_conversion_fraction is None
 
     def test_modifiying_gene_conversion(self):
-        genome = self.species.get_contig(
-            chromosome="Chromosome", gene_conversion_rate=0.1, gene_conversion_length=1
-        )
-        assert genome.gene_conversion_rate == 0.1
-        assert genome.gene_conversion_length == 1
-
-    def test_use_species_gene_conversion(self):
-        genome = self.species.get_contig(
-            chromosome="Chromosome", use_species_gene_conversion=True
-        )
-        assert genome.gene_conversion_rate == 8.9e-11
-        assert genome.gene_conversion_length == 345
-
-    def test_unnamed_contig_with_gc(self):
-        genome = self.species.get_contig(
-            length=42, gene_conversion_rate=0.1, gene_conversion_length=1
-        )
-        assert genome.gene_conversion_rate == 0.1
-        assert genome.gene_conversion_length == 1
-
-    def test_use_species_gene_conversion_unnamed_contig(self):
-        genome = self.species.get_contig(length=42, use_species_gene_conversion=True)
-        assert genome.gene_conversion_rate == 8.9e-11
-        assert genome.gene_conversion_length == 345
+        esc_col = stdpopsim.get_species("EscCol")
+        assert esc_col.genome.mean_gene_conversion_fraction == 0
+        contig = esc_col.get_contig(chromosome="Chromosome")
+        assert contig.bacterial_recombination is True
+        assert contig.gene_conversion_fraction is None
+        assert contig.gene_conversion_length > 0
+
+        rm = msprime.RateMap(position=[0.0, contig.length], rate=[0.1])
+        contig.recombination_map = rm
+        contig.gene_conversion_length = 1
+        assert contig.recombination_map == rm
+        assert contig.gene_conversion_length == 1
```

Comment:
maybe i'm missing it, but it seems like you should have a test that checks if recombination_map of a contig grows when the gene_conversion_fraction is > 0